### ISSUE TITLE
e2e flake: should be able to up and down services: Only 2 pods started out of 3 

### ISSUE THREAD

**Issue by jsafrane** [kubernetes/kubernetes#25461]
Logs: https://console.cloud.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/25216/kubernetes-pull-build-test-e2e-gce/39285/

from build-log.txt:
```
[It] should be able to up and down services
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/service.go:277


[snip]

May 10 08:03:50.678: INFO: service2 Pods: 3 out of 3 created, 2 running, 1 pending, 0 waiting, 0 inactive, 0 terminating, 0 unknown, 0 runningButNotReady 
May 10 08:03:50.678: INFO: Pod service2-z4s7y assigned to host e2e-gce-builder-3-3-minion-9my2 (IP: 10.240.0.5) in Pending


[snip]

May 10 08:03:50.895: INFO: At {2016-05-10 08:01:37 -0700 PDT} - event for service2-z4s7y: {kubelet e2e-gce-builder-3-3-minion-9my2} Pulling: pulling image "gcr.io/google_containers/serve_hostname:v1.4"
May 10 08:03:50.895: INFO: At {2016-05-10 08:03:37 -0700 PDT} - event for service2-z4s7y: {kubelet e2e-gce-builder-3-3-minion-9my2} FailedSync: Error syncing pod, skipping: failed to "StartContainer" for "service2" with ErrImagePull: "operation timeout: context deadline exceeded"

May 10 08:03:50.895: INFO: At {2016-05-10 08:03:37 -0700 PDT} - event for service2-z4s7y: {kubelet e2e-gce-builder-3-3-minion-9my2} Failed: Failed to pull image "gcr.io/google_containers/serve_hostname:v1.4": operation timeout: context deadline exceeded

May 10 08:03:50.895: INFO: At {2016-05-10 08:03:38 -0700 PDT} - event for service2-z4s7y: {kubelet e2e-gce-builder-3-3-minion-9my2} BackOff: Back-off pulling image "gcr.io/google_containers/serve_hostname:v1.4"
May 10 08:03:50.895: INFO: At {2016-05-10 08:03:38 -0700 PDT} - event for service2-z4s7y: {kubelet e2e-gce-builder-3-3-minion-9my2} FailedSync: Error syncing pod, skipping: failed to "StartContainer" for "service2" with ImagePullBackOff: "Back-off pulling image \"gcr.io/google_containers/serve_hostname:v1.4\""
```

It may be dup of #21029

**Comment by Random-Liu** [comment]
Cool, at least this proves that timeout works...@yujuhong

**Comment by yujuhong** [comment]
Let's consolidate the image pulling issues and discuss a generic fix in #25461
